src/lib/visualization/abstract_visualizer.hpp

```diff
@@ -188,6 +211,9 @@ class AbstractVisualizer {
   VizGraphInfo _graph_info;
   VizVertexInfo _default_vertex;
   VizEdgeInfo _default_edge;
+
+  // Current index of color in _random_color()
+  size_t _random_color_index{0};
```

Comment:
static variable within the color picker function?

Reply:
had it like that before, didn't like it. A tad more determinism the way it is done now.